### ISSUE TITLE
Using mark_sharding vs. MpDeviceLoader with input_sharding=xs.ShardingSpec

### ISSUE THREAD

**Issue by dudulightricks** [pytorch/xla#7854]
## ❓ Questions and Help
If we have a few tensors in a batch with different sizes and we use mark_sharding on each of them, we lose something comparing to input_sharding=xs.ShardingSpec in the MpDeviceLoader (which only works for a single size of tensor in the batch)? @JackCaoG

**Comment by JackCaoG** [comment]
@alanwaketan can you take this one?

**Comment by alanwaketan** [comment]
Can you pad your tensors?

**Comment by dudulightricks** [comment]
@alanwaketan Yes I can but I'm asking if it would give me any benefit compared to just mark_sharding on each tensor.
